freeciv-web/src/main/webapp/javascript/pregame.js

```diff
@@ -1426,27 +1426,18 @@ function create_new_freeciv_user_account_request(action_type)
   var email = $("#email").val().trim();
   var captcha = $("#g-recaptcha-response").val();
 
-  var cleaned_username = username.replace(/[^a-zA-Z]/g,'');
-
   $("#username_validation_result").show();
+  var invalid_name = get_invalid_username_reason(username);
+  if (invalid_name != null) {
+    $("#username_validation_result").html("Your name is " + invalid_name + ".");
```

Comment:
Pedantic semantics (:grin:): for users, maybe `("The username '" + username + "' is " + invalid_name + ".")` is slightly more clear, and gives feedback as to the name which failed. In short, it's not their username yet! Another option: `"Your choice is "...`. I favor the first.

Reply:
Adding the name to the message is a good idea. Not a big difference here, but in hotseat, with more than one user in the same form...